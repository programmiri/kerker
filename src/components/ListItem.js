import React from "react";
import PropTypes from "prop-types";
import { dateToString } from "../lib/format.js";

function ListItem(props) {

  const handleClick = event => {
    event.preventDefault();
    props.setCurrentNote(props.note);
    props.setCurrentNote(props.note);
    props.setCurrentNoteEncodingState(false);
  };

  const textIsOpen = props.isActive ? 'is open' : null;
  const activeItemClass = props.isActive ? 'ListItem-active-item' : null;
  const backgroundClass = props.index % 2 === 0 ? 'bg-light' : null;

  return (
    <li
      className={`ListItem list-group-item p-2 ${backgroundClass} ${activeItemClass}`}
      key={`${props.id}`}
      onClick={handleClick}
    >
      <div className="ListItem-open-hint font-s text-secondary text-right">
        {textIsOpen}
      </div>
      <p className="ListItem-title my-1">{props.title}</p>
      <small className="ListItem-date text-dark font-s font-monospace mb-2">
        created: {dateToString(props.createdAt)}
      </small>
    </li>
  );
}

export default ListItem;

ListItem.defaultProps = {
  isActive: true
}

ListItem.propTypes = {
  id: PropTypes.string.isRequired,
  title: PropTypes.string.isRequired,
  createdAt: PropTypes.instanceOf(Date).isRequired,
  isActive: PropTypes.bool.isRequired,
};
