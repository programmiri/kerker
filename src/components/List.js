import React from "react";
import ListItem from "./ListItem";
import PropTypes from "prop-types";

function List(props) {
  const listItems = props.notes.map((item, index) => (
    <ListItem
      key={item.id}
      id={item.id}
      title={item.title}
      createdAt={item.createdAt}
      note={item}
      setCurrentNote={props.setCurrentNote}
      setCurrentNoteEncodingState={props.setCurrentNoteEncodingState}
      index={index}
      isActive={props.currentNote === item}
    />
  ));
  return <ul className="List list-group">{listItems}</ul>;
}

export default List;

List.propTypes = {
  notes: PropTypes.array.isRequired
};
