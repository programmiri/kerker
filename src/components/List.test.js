import React from "react";
import { shallow } from "enzyme";
import toJson from "enzyme-to-json";

import List from "./List";
import ListItem from "./ListItem";

const notesFixtures = [
  {
    id: "109156be-c4fb-41ea-b1b4-efe1671c5836",
    title: "Hello",
    body: "This is my body",
    bodyPersistedVersion: "This is my body",
    bodyEncrypted: "SECRET",
    createdAt: new Date(1517743704 * 1000),
    updatedAt: new Date(1517743704 * 1000)
  },
  {
    id: "209156be-c4fb-41ea-b1b4-efe1671c5836",
    title: "Another note",
    body: "Another body",
    bodyPersistedVersion: "Another body",
    bodyEncrypted: "SECRET2",
    createdAt: new Date(1517743704 * 1000),
    updatedAt: new Date(1517743704 * 1000)
  }
];

it("renders without crashing", () => {
  shallow(<List notes={notesFixtures} />);
});

describe("renders a list of notes", () => {
  const wrapper = shallow(<List notes={notesFixtures} />);

  it("renders an ul with the class list-group", () => {
    expect(wrapper.find("div").hasClass("list-group")).toEqual(true);
  });

  it("renders every note as <ListItem />", () => {
    expect(wrapper.find(ListItem).length).toEqual(notesFixtures.length);
  });

  it("snapshot test", () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
