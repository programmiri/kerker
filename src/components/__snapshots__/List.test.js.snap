// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders a list of notes snapshot test 1`] = `
<div
  className="List list-group"
>
  <ListItem
    createdAt={2018-02-04T11:28:24.000Z}
    id="109156be-c4fb-41ea-b1b4-efe1671c5836"
    key="109156be-c4fb-41ea-b1b4-efe1671c5836"
    note={
      Object {
        "body": "This is my body",
        "bodyEncrypted": "SECRET",
        "bodyPersistedVersion": "This is my body",
        "createdAt": 2018-02-04T11:28:24.000Z,
        "id": "109156be-c4fb-41ea-b1b4-efe1671c5836",
        "title": "Hello",
        "updatedAt": 2018-02-04T11:28:24.000Z,
      }
    }
    title="Hello"
  />
  <ListItem
    createdAt={2018-02-04T11:28:24.000Z}
    id="209156be-c4fb-41ea-b1b4-efe1671c5836"
    key="209156be-c4fb-41ea-b1b4-efe1671c5836"
    note={
      Object {
        "body": "Another body",
        "bodyEncrypted": "SECRET2",
        "bodyPersistedVersion": "Another body",
        "createdAt": 2018-02-04T11:28:24.000Z,
        "id": "209156be-c4fb-41ea-b1b4-efe1671c5836",
        "title": "Another note",
        "updatedAt": 2018-02-04T11:28:24.000Z,
      }
    }
    title="Another note"
  />
</div>
`;
